core/templates/services/csrf-token.service.spec.ts

```diff
@@ -21,33 +21,53 @@ describe('Csrf Token Service', function() {
   let csrfTokenService: CsrfTokenService;
   beforeEach(() => {
     csrfTokenService = new CsrfTokenService();
-    spyOn(window, 'fetch').and.callFake((reqInfo: RequestInfo) => {
-      return Promise.resolve(
-        new Response('12345{"token": "sample-csrf-token"}')
-      );
-    });
   });
 
-  it('should correctly set the csrf token', (done) => {
-    csrfTokenService.initializeToken();
+  describe('when fetch succeeds', function() {
+    beforeEach(() => {
+      spyOn(window, 'fetch').and.callFake((reqInfo: RequestInfo) => {
+        return Promise.resolve(
+          new Response('12345{"token": "sample-csrf-token"}')
+        );
+      });
+    });
 
-    csrfTokenService.getTokenAsync().then(function(token) {
-      expect(token).toEqual('sample-csrf-token');
-    }).then(done, done.fail);
-  });
+    it('should correctly set the csrf token', (done) => {
+      csrfTokenService.initializeToken();
 
-  it('should error if initialize is called more than once', () => {
-    csrfTokenService.initializeToken();
+      csrfTokenService.getTokenAsync().then(function(token) {
+        expect(token).toEqual('sample-csrf-token');
+      }).then(done, done.fail);
+    });
+
+    it('should error if initialize is called more than once', () => {
+      csrfTokenService.initializeToken();
 
-    expect(() => csrfTokenService.initializeToken())
-      .toThrowError('Token request has already been made');
+      expect(() => csrfTokenService.initializeToken())
+        .toThrowError('Token request has already been made');
+    });
+
+    it('should error if getTokenAsync is called before initialize', () => {
+      try {
+        csrfTokenService.getTokenAsync();
+      } catch (e) {
+        expect(e).toEqual(new Error('Token needs to be initialized'));
+      }
+    });
   });
 
-  it('should error if getTokenAsync is called before initialize', () => {
-    try {
-      csrfTokenService.getTokenAsync();
-    } catch (e) {
-      expect(e).toEqual(new Error('Token needs to be initialized'));
-    }
+  describe('when fetch fails', function() {
+    beforeEach(() => {
+      spyOn(window, 'fetch').and.rejectWith(new Error('Test Error'));
+    });
+
+    it('should result in a handled promise rejection', () => {
+      try {
+        csrfTokenService.initializeToken();
+      } catch (e) {
+        expect(e).toEqual(new Error(
+          'THere has been a problem with the fetch operation: Test Error'));
```

Comment:
```suggestion
          'There has been a problem with the fetch operation: Test Error'));
```

Interesting that the test passes, maybe not case sensitive.

Reply:
Thanks! I fixed it.